### ISSUE TITLE
plaintext password in globally readable file

### ISSUE THREAD

**Issue by KenKundert** [sparrowwallet/sparrow#121]
The password for the bitcoin node is saved in the config file in plain text. By default this file and the directory that contains it are readable by all.

At a minimum the file and directory should be readable only by the user.  Better yet would be to encrypt the password.

**Comment by craigraw** [comment]
> At a minimum the file and directory should be readable only by the user. 

Implemented in a0d3f3b, creation of the config file is now owner read/write only. 

> Better yet would be to encrypt the password.

Without the user needing to supply an application level password, such encryption would be trivial to circumvent.

**Comment by KenKundert** [comment]
> Without the user needing to supply an application level password, such encryption would be trivial to circumvent.

Agreed.  It would be great to have an application level password so that information like this can be safely concealed.  I would especially like the number and names of wallets concealed.  Even if the password were optional, it would be available to provide security to those that want it while not encumbering those that do not feel the need for it.

**Comment by craigraw** [comment]
I'll consider this, but I'd encourage you to consider running an Electrum server instead. Connecting directly to Bitcoin Core uses the Bitcoin Core wallet which is inherently insecure - the RPC password is transmitted over clear text, and the Bitcoin Core watch only wallet itself uses no encryption on the public keys/addresses. Using an Electrum server avoids all of these issues. Sparrow itself uses very good key derivation and encryption algorithms to protect the wallets it stores.

**Comment by KenKundert** [comment]
I decided to take your advice and try Electrum Server.  Upon diving in I found it to be a significant amount of effort so I decided to try Electrum Personal Server, which I already had running.  It seems to work with Sparrow without difficulty, and there was no sign of the wallet or transactions showing up in Bitcoin Core.  You might consider mentioning EPS in your documentation as a light-weight alternative to Electrum Server.

My goal here was get some kind of plausible deniability as to the presence of my wallets.  But as I thought it through a bit more, I realized that concealing the names and number of wallets in the wallets directory is just one of several things that must be done to provide plausible deniability, and you don't gain much unless you do all of them.  In hindsight, your initial suggestion of using removable storage for the configuration directory is the best approach.  I am afraid I have wasted your time, and for that I apologize.

I would also like to thank you for your patience and thoughtful responses.  I learned quite a bit from the exchange.
